doc/design/backup_operator.md

```diff
@@ -0,0 +1,63 @@
+# etcd Backup Operator Design Document
+
+## Problem
+
+Currently, etcd-operator supports various backup strategies such as PV, S3, and Azure Blob Store. However, those strategies are embedded into the etcd operator code itself. Support for new backup strategy are required to be pushed into etcd-operator codebase. As more backup strategies are added, the etcd-operator code base would be hard to maintain. In Addition, users may want to implement their backup strategy without confining to the current etcd-operator’s backup spec and may not want to dependent on etcd-operator codebase for backup.
+
+## Solution
+
+To give user the flexibility of handling etcd backup his/her own way and to decouple backup logic out of etcd operator, a standalone etcd-backup operator can be used to manage the backup.
+
+## A Proposed Architecture 
+
+In order to control backup of etcd clusters, a standalone etcd backup operator can be used to listen for a customized backup CRD; it specifies the detail of etcd cluster backup. Then, etcd backup operator spawns and manages a sidecar that can communicate to the etcd cluster and create backup when needed.
+
+The etcd backup operator business logic:
+
+* Create a backup CRD e.g:
+```yaml
+apiVersion: apiextentions.k8s.io/v1beta1
+kind: CustomResourceDefiniton
+metadata:
+  name: etcdbackups.etcd.database.coreos.com
+spec:
+  group: etcd.database.coreos.com
+  version: v1alpha1
+  Scope: Namespaced
+  names: 
+    kind: EtcdBackup
+    plural: etcdbackups
+```
+
+* Create an informer that listens on the backup CRD
+Create/manage a backup sidecar pod for each new CR:
+For instance, the new CR for S3 backup:
+
+```yaml
+apiVersion: "etcd.database.coreos.com/v1alpha1"
```

Comment:
a backup should have a status field to indicate if the backup is created or failed, etc..

Reply:
To keep it aligned with etcd operator's group version:
```
.../v1beta2
```